packages/beacon-node/src/execution/engine/http.ts

```diff
@@ -575,6 +558,45 @@ type EngineRequestKey = keyof EngineApiRpcParamTypes;
 type EngineRequestByKey = {
   [K in EngineRequestKey]: {method: K; params: EngineApiRpcParamTypes[K]; methodOpts: ReqOpts};
 };
-type EngineRequest = EngineRequestByKey[EngineRequestKey];
+export type EngineRequest = EngineRequestByKey[EngineRequestKey];
 type EngineResponseByKey = {[K in EngineRequestKey]: EngineApiRpcReturnTypes[K]};
 type EngineResponse = EngineResponseByKey[EngineRequestKey];
+
+/**
+ *
+ * Generate serialized params according to `method`.
+ *
+ * `params` must not allow numeric keys because we rely Object.keys() to return keys in insertion order.
+ * eg. {versionedHashes, parentBlockRoot, executionRequests} has to return [versionedHashes, parentBlockRoot, executionRequests]
+ * Having numeric key will not guarantee keys in insertion order.
+ */
+export function generateSerializedNewPayloadParams<T extends EngineNewPayloadMethod>(
+  fork: ForkName,
+  method: T,
+  params: {[K in keyof EngineApiParamTypes[T]]: EngineApiParamTypes[T][K] | undefined}
```

Comment:
```suggestion
  params: RequireAtLeastOne<EngineApiParamTypes[T]>
```

I think this will be sufficient to get same type and the usage more type safe. 

with the utility type

```ts
type RequireAtLeastOne<T, Keys extends keyof T = keyof T> =
  // First, keep properties not in Keys (if any)
  Pick<T, Exclude<keyof T, Keys>> &
  // Then for the keys in Keys, create a union where each key is required and the rest optional.
  {
    [K in Keys]-?: Required<Pick<T, K>> & Partial<Pick<T, Exclude<Keys, K>>>
  }[Keys];
```

Reply:
> I think this will be sufficient to get same type and the usage more type safe.
> 
> with the utility type
> 
> ```ts
> type RequireAtLeastOne<T, Keys extends keyof T = keyof T> =
>   // First, keep properties not in Keys (if any)
>   Pick<T, Exclude<keyof T, Keys>> &
>   // Then for the keys in Keys, create a union where each key is required and the rest optional.
>   {
>     [K in Keys]-?: Required<Pick<T, K>> & Partial<Pick<T, Exclude<Keys, K>>>
>   }[Keys];
> ```

I am not sure what `RequireAtLeastOne` is trying to accomplish. Is it specifically to reject empty object? Because the current `params` definition already reject this case. 

Also I think `RequireAtLeastOne` is too complicated, and I am not sure where to put it.